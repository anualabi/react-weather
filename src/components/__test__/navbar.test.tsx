import { render, screen } from "@testing-library/react";

import Navbar from "@/components/navbar";

describe("Navbar", () => {
  it("should render (snapshot)", () => {
    const { asFragment } = render(<Navbar />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render an HTML nav element", () => {
    render(<Navbar />);
    const navbarElement = screen.getByRole("navigation");
    expect(navbarElement.tagName).toBe("NAV");
  });

  it("should contain the Logo component's content", () => {
    render(<Navbar />);
    const logoText = screen.getByText("React Weather");
    expect(logoText).toBeInTheDocument();
  });
});
