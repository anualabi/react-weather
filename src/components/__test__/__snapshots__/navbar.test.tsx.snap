// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render (snapshot) 1`] = `
<DocumentFragment>
  <nav
    class="bg-secondary h-16 md:h-20 xl:h-24 flex items-center"
  >
    <div
      class="mx-auto w-full max-w-[1440px] px-5 xl:px-11"
    >
      <a
        href="/"
      >
        <h2
          class="scroll-m-20 text-2xl md:text-3xl xl:text-4xl font-black capitalize text-white"
        >
          React Weather
        </h2>
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
